patches/minecraft/net/minecraft/world/chunk/Chunk.java.patch

```diff
@@ -115,15 +115,30 @@
                          this.field_76637_e.func_175690_a(p_177436_1_, tileentity1);
                      }
  
-@@ -738,6 +736,7 @@
+@@ -738,11 +736,13 @@
              k = this.field_76645_j.length - 1;
          }
  
 +        net.minecraftforge.common.MinecraftForge.EVENT_BUS.post(new net.minecraftforge.event.entity.EntityEvent.EnteringChunk(p_76612_1_, this.field_76635_g, this.field_76647_h, p_76612_1_.field_70176_ah, p_76612_1_.field_70164_aj));
          p_76612_1_.field_70175_ag = true;
          p_76612_1_.field_70176_ah = this.field_76635_g;
          p_76612_1_.field_70162_ai = k;
-@@ -778,7 +777,7 @@
+         p_76612_1_.field_70164_aj = this.field_76647_h;
+         this.field_76645_j[k].add(p_76612_1_);
++        this.func_76630_e(); // Forge - ensure chunks are marked to save after an entity add
+     }
+ 
+     public void func_76622_b(Entity p_76622_1_)
+@@ -762,7 +762,7 @@
+             p_76608_2_ = this.field_76645_j.length - 1;
+         }
+ 
+-        this.field_76645_j[p_76608_2_].remove(p_76608_1_);
++        if (!this.field_76645_j[p_76608_2_].remove(p_76608_1_)) this.func_76630_e(); // Forge - ensure chunks are marked to save after entity removals only
```

Comment:
don't think you need the if statement here, but I think you have it inverted anyways? This is only marking if it failed to remove the entity.

Reply:
From the paper source, ti looks like it only called to update the field if and only if the entity removal was successful (yes, there's still a logical flip that needs to be corrected).

Reply:
Yeah but that was because we have other operations being done in that method increasing/decreasing counts, that we want to ensure only runs when something actually changed state.